panel/panel.py

```diff
@@ -207,107 +175,103 @@ def fill_panel(window, update=False, **panel_filter):
     else:
         settings.set("sublime_linter_panel_filter", panel_filter)
 
-    to_render = RenderLines()
-    for vid, view_dict in errors.items():
-        if util.is_none_or_zero(view_dict["we_count_view"]):
-            continue
-
-        view_lines = get_view_lines(view_dict, panel_filter)
-        if view_lines:
+    to_render = []
+    for bid, buf_dict in errors.items():
+        buf_lines = get_buf_lines(buf_dict, panel_filter)
+        if buf_lines:
             # append header
-            to_render.append(format_header(path_dict[vid]))
+            to_render.append(format_header(path_dict[bid]))
 
             # append lines
-            for lineno, error_type, item in view_lines:
-                to_render.append(format_row(lineno, error_type, item))
-                item["panel_lineno"] = to_render.current_lineno()
+            base_lineno = len(to_render)
+            for i, item in enumerate(buf_lines):
+                to_render.append(format_row(item))
+                item["panel_line"] = base_lineno + i
```

Comment:
mutate alert!

Reply:
Okay if it's too hard, skip this comment

Reply:
We need that for updating the panel selection, that you all adore so much.
It might have escaped your notice that the panel line was written to `persist.errors` also.
If you got a different suggestion, just tell.

Reply:
I don't tell you yet bc than you freak out and change everything again.

Reply:
I am eager to find out what you got in your sleeves.